docs/source/reference/optimizer_hooks.rst

```diff
@@ -0,0 +1,7 @@
+Hook functions
```

Comment:
I think `Optimizer hook functions` or `Optimizer hooks` is more appropriate to avoid confusion with `chainer.function_hooks`.